### PR TITLE
[SYCL][NewOffload][E2E] add a single test for --offload-new-driver

### PR DESCRIPTION
**Pull request by jasonlizhengjian**
A single basic file to compile and run to test functionality of --offload-new-driver

### CODE REVIEW

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (asudarsa) [inline sycl/test-e2e/NewOffloadDriver/buffer.cpp:5]: We try to use a subset here. Please do:
```suggestion
#include <sycl/detail/core.hpp>
```

**Comment by asudarsa** [comment]
Currently, we support only JIT compilation flows in SYCL offloading using new offload model. This test will fail for AOT compilations.

Thanks

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (sarnex) [inline sycl/test-e2e/NewOffloadDriver/buffer.cpp:2]: to force JIT i think we can do
`%clangxx -fsycl --offload-new-driver ...`
because if it's set up for AOT, `%{build}` will expand to AOT options

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (sarnex) [inline sycl/test-e2e/NewOffloadDriver/buffer.cpp:2]: i think we need `-fsycl` here

**Comment by sarnex** [comment]
LGTM, will approve after CI passes

**Review by asudarsa — APPROVED** [review]
Looks good to me. Couple of nits. (1) it is a good idea to add a comment at the top mentioning what the test does. I am ok to start doing this from the next PR. and (2) EOF seems to be missing a linebreak. 
(2) can be addressed just before merging (after CI testing is complete).

Thanks again for adding this.

**Comment by sarnex** [comment]
@jasonlizhengjian I think we just have to fix the formatting and then we are ready to merge!

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (sarnex) [inline sycl/test-e2e/NewOffloadDriver/buffer.cpp:3]: These should be below the copyright header

**Review by sarnex — APPROVED** [review]

**Comment by sarnex** [comment]
@intel/llvm-reviewers-runtime Can someone please take a look at this? It's actually testing a dpcpp tools feature so should not be a difficult review! Thanks!

**Review by uditagarwal97 — APPROVED** [review]

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (sarnex) [reply sycl/test-e2e/NewOffloadDriver/buffer.cpp:11]: yes it's correct, this is testing a new device linking mode that currently only supports intel GPUs with JIT, so we are restricting the test to match that

File: sycl/test-e2e/NewOffloadDriver/buffer.cpp
Comment (sarnex) [reply sycl/test-e2e/NewOffloadDriver/buffer.cpp]: imo this pr has been through a lot of iteration already, we should resolve this question in general for all tests outside of this pr and then make a changing implementing the rule (and ideally a CI check to enforce it in the future

**Review by maarquitos14 — APPROVED** [review]

**Pull request merged**
